pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -87,20 +88,47 @@ func (s *statesInformer) calGuaranteedCpu(usedCPUs map[int32]*extension.CPUInfo,
 	if err != nil {
 		return nil, err
 	}
-	podAllocs := []extension.PodCPUAlloc{}
-	for pod := range checkPoint.Entries {
+
+	pods := s.GetAllPods()
+	managedPods := make(map[types.UID]struct{})
+	for _, podMeta := range pods {
+		qosClass := extension.GetPodQoSClass(podMeta.Pod)
+		if qosClass == extension.QoSLS || qosClass == extension.QoSBE {
+			managedPods[podMeta.Pod.UID] = struct{}{}
+			continue
+		}
+		resourceStatus, err := extension.GetResourceStatus(podMeta.Pod.Annotations)
+		if err == nil {
+			set, err := cpuset.Parse(resourceStatus.CPUSet)
+			if err == nil && set.Size() > 0 {
+				managedPods[podMeta.Pod.UID] = struct{}{}
+			}
+		}
+	}
+
+	var podAllocs []extension.PodCPUAlloc
+	for podUID := range checkPoint.Entries {
+		if _, ok := managedPods[types.UID(podUID)]; ok {
+			continue
+		}
 		cpuSet := cpuset.NewCPUSet()
-		for container, cpuString := range checkPoint.Entries[pod] {
-			if tmpContainerCPUSet, err := cpuset.Parse(cpuString); err != nil {
-				klog.Errorf("could not parse cpuset %q for container %q in pod %q: %v", cpuString, container, pod, err)
+		for container, cpuString := range checkPoint.Entries[podUID] {
+			if containerCPUSet, err := cpuset.Parse(cpuString); err != nil {
+				klog.Errorf("could not parse cpuset %q for container %q in pod %q: %v", cpuString, container, podUID, err)
 				continue
-			} else {
-				cpuSet = cpuSet.Union(tmpContainerCPUSet)
+			} else if containerCPUSet.Size() > 0 {
+				cpuSet = cpuSet.Union(containerCPUSet)
 			}
 		}
+		if cpuSet.IsEmpty() {
+			continue
+		}
+		// TODO: It is possible that the data in the checkpoint file is invalid
+		//  and should be checked with the data in the cgroup to determine whether it is consistent
 		podAllocs = append(podAllocs, extension.PodCPUAlloc{
-			Name:   pod,
-			CPUSet: cpuSet.String(),
+			UID:              types.UID(podUID),
+			CPUSet:           cpuSet.String(),
+			ManagedByKubelet: true,
```

Comment:
how about adding pod ns/name for better readability

Reply:
updated